packages/website/docs/templates.md

```diff
@@ -0,0 +1,277 @@
+
+:::
+
+### Returning HTML
+
+Native HTML elements aren't valid VNodes, which means you can't return a template string that contains HTML, or an HTML element. But if you're not using a virtual DOM implementation in your app, there are still two ways you can return HTML.
+
+#### Using `createElement`
+
+Each template function provides access to `createElement` and `Fragment` to create VNodes.
+
+```js
+import { autocomplete } from '@algolia/autocomplete-js';
+
+autocomplete({
+  // ...
+  getSources() {
+    return [
+      {
+        // ...
+        templates: {
+          item({ item, createElement, Fragment }) {
+            return createElement(Fragment, {}, item.name);
+          },
+        },
+      },
+    ];
+  },
+});
+```
+
+You can also leverage `dangerouslySetInnerHTML` if you need to provide more complex HTML without having to create nested VNodes.
+
+```js
+import { autocomplete } from '@algolia/autocomplete-js';
+
+autocomplete({
+  // ...
+  getSources() {
+    return [
+      {
+        // ...
+        templates: {
+          item({ item, createElement }) {
+            return createElement('div', {
+              dangerouslySetInnerHTML: {
+                __html: `<div>
+                  <img
+                    src="${item.image}"
+                    alt="${item.name}"
+                  />
+                </div>
+                <div>
+                  ${item.name}
+                </div>`,
+              },
+            });
+          },
+        },
+      },
+    ];
+  },
+});
+```
+
+By default, `createElement` and `Fragment` default to [Preact](https://preactjs.com/)'s `Preact.createElement` (or `h`) and `preact.Fragment`. You can customize these and provide the virtual DOM implementation you prefer.
+
+#### Using the `htm` library
+
+If you're not using a transpiler to build your app, you can still use Autocomplete with the [htm](https://github.com/developit/htm) library, which lets you use a JSX-like syntax directly in the browser.
+
+```jsx
+import { html } from 'htm/preact';
+import { autocomplete } from '@algolia/autocomplete-js';
+
+autocomplete({
+  // ...
+  getSources() {
+    return [
+      {
+        // ...
+        templates: {
+          item({ item }) {
+            return html`<div>${item.name}</div>`
+          },
+        },
+      },
+    ];
+  },
+});
+```
+
+### Rendering a header and footer
+
+In addition to rendering items, you can customize what to display before and after the list of items using the `header` and `footer` templates.
+
+```js
+autocomplete({
+  // ...
+  getSources({ query }) {
+    return [
+      {
+        // ...
+        templates: {
+          header() {
+            return 'Suggestions';
+          },
+          item({ item }) {
+            return `Result: ${item.name}`;
+          },
+          footer() {
+            return 'Footer';
+          },
```

Comment:
Since #395 there's also an `empty` template that is called instead of `item` if the source's items are empty. I think we can document it in this PR.

Reply:
Sweet! In the PR it's still using the former API with the DOM implementation, but I assume it will have the same signature as `header` and `footer` (minus `items`), correct?

Reply:
Moving on with this assumption, addressed in 3ba55fe.